Framework/Geometry/src/Instrument/StructuredDetector.cpp

```diff
@@ -0,0 +1,701 @@
+  shapestr << "</bounding-box>";
+  shapestr << "<algebra val=\"" << name << "\" />";
+  shapestr << "</type>";
+
+  boost::shared_ptr<Mantid::Geometry::Object> shape =
+      shapeCreator.createShape(shapestr.str(), false);
+
+  // Create detector
+  auto detector = new Detector(name, id, shape, parent);
+
+  // Set detector position relative to parent
+  V3D pos(xpos, ypos, 0);
+
+  detector->translate(pos);
+
+  return detector;
+}
+
+/** Returns the minimum detector id
+* @return minimum detector id
+*/
+int StructuredDetector::minDetectorID() {
+  if (m_map)
+    return m_base->m_minDetId;
+  return m_minDetId;
+}
+
+//-------------------------------------------------------------------------------------------------
+/** Returns the maximum detector id
+* @return maximum detector id
+*/
+int StructuredDetector::maxDetectorID() {
+  if (m_map)
+    return m_base->m_maxDetId;
+  return m_maxDetId;
+}
+
+boost::shared_ptr<const IComponent>
+StructuredDetector::getComponentByName(const std::string &cname,
+                                       int nlevels) const {
+  // exact matches
+  if (cname == this->getName())
+    return boost::shared_ptr<const IComponent>(this);
+
+  // cache the detector's name as all the other names are longer
+  // The extra ( is because all children of this have that as the next character
+  // and this prevents Bank11 matching Bank 1
+  const std::string MEMBER_NAME = this->getName() + "(";
+
+  // if the component name is too short, just return
+  if (cname.length() <= MEMBER_NAME.length())
+    return boost::shared_ptr<const IComponent>();
+
+  // check that the searched for name starts with the detector's
+  // name as they are generated
+  if (cname.substr(0, MEMBER_NAME.length()).compare(MEMBER_NAME) == 0) {
+    return CompAssembly::getComponentByName(cname, nlevels);
+  } else {
+    return boost::shared_ptr<const IComponent>();
+  }
+}
+
+//-------------------------------------------------------------------------------------------------
+//-------------------------------------------------------------------------------------------------
+// ------------ IObjComponent methods ----------------
+//-------------------------------------------------------------------------------------------------
+//-------------------------------------------------------------------------------------------------
+
+//-------------------------------------------------------------------------------------------------
+/// Does the point given lie within this object component?
+bool StructuredDetector::isValid(const V3D &point) const {
+  // Avoid compiler warning
+  (void)point;
+  throw Kernel::Exception::NotImplementedError(
+      "StructuredDetector::isValid() is not implemented.");
+}
+
+/// Does the point given lie on the surface of this object component?
+bool StructuredDetector::isOnSide(const V3D &point) const {
+  // Avoid compiler warning
+  (void)point;
+  throw Kernel::Exception::NotImplementedError(
```

Comment:
Is there a reason we don't implement this? Is this to do with the StructuredDetector being infinitely thin?